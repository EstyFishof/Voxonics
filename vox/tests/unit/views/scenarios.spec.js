import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router';
import ParentScenario from '@/views/Config/scenarios2/ParentScenario.vue';
import Item from '@/views/Config/scenarios2/Item.vue';
import dialConfiguration from '@/views/Config/scenarios2/components/dialConfiguration.vue';
import queueConfiguration from '@/views/Config/scenarios2/components/queueConfiguration.vue';
import vNumberConfiguration from '@/views/Config/scenarios2/components/vNumberConfiguration.vue';
import mock from './__mock__/scenario';
import VueTippy, { TippyComponent } from "vue-tippy";

const localVue = createLocalVue();
localVue.use(Vuex);
const router = new VueRouter();
localVue.use(Vuex);
localVue.use(VueTippy);
localVue.component("tippy", TippyComponent);

describe('Scenario', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };
    const store = new Vuex.Store({
        actions: {
            setListScenarios({commit}, value) {},
            setScenario({commit}, value) {},
            setShowConfig({commit}, value) {},
            setItemScenario({commit}, value) {},
            setChangeNameScenario({commit}, value) {},
            changeItemScenarios({commit}, value) {}
        },
        getters: {
            getListScenarios: () => mock.getListScenarios,
            getScenario: () => mock.getScenario,
            getShowConfig: () => mock.getShowConfig,
            getInitialList: () => mock.getInitialList
        }
    });

    const wrapper = shallowMount(ParentScenario, { localVue, router, store, stubs: {ParentScenario: true} });
    it('it should  Vue instance', () => {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('setScenario toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setScenario');
        await wrapper.vm.setScenario(mock.getScenario);
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.edit).toStrictEqual(false);
        expect(wrapper.vm.activeTab).toStrictEqual('/ewrwe');
        spy.mockRestore();
    });
    it('changeName toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'changeName');
        await wrapper.vm.changeName('123');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setName toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setName');
        await wrapper.vm.setName('123');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('deleteScenario toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'deleteScenario');
        await wrapper.vm.deleteScenario(mock.getScenario, 'parent');
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.deleteScenarios).toStrictEqual(['3d3c5b96-8694-470d-85ad-f3e43522ff68']);
        expect(wrapper.vm.edit).toStrictEqual(true);
        spy.mockRestore();
    });
    it('deleteScenario toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'deleteScenario');
        await wrapper.vm.deleteScenario({path: '/ewrwe/name', uid: 'delete'}, 'child');
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.deleteScenarios).toStrictEqual(['3d3c5b96-8694-470d-85ad-f3e43522ff68', 'delete']);
        expect(wrapper.vm.edit).toStrictEqual(true);
        spy.mockRestore();
    });
    it('setNewScenario toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setNewScenario');
        await wrapper.vm.setNewScenario('name');
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.activeTab).toStrictEqual('/name');
        expect(wrapper.vm.name.mainName).toStrictEqual('');
        expect(wrapper.vm.name.secondaryName).toStrictEqual('');
        spy.mockRestore();
    });
    it('clearChanges toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'clearChanges');
        await wrapper.vm.clearChanges();
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.deleteScenarios).toStrictEqual([]);
        expect(wrapper.vm.edit).toStrictEqual(false);
        spy.mockRestore();
    });
    it('applyChanges toHaveBeenCalled', async () => {
        await wrapper.setData({deleteScenarios: [{}, {}]});
        const spy = jest.spyOn(wrapper.vm, 'applyChanges');
        await wrapper.vm.applyChanges();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('applyChanges toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'applyChanges');
        await wrapper.vm.applyChanges();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('applyChanges toHaveBeenCalled', async () => {
        await wrapper.setData({getScenario: {uid: '111'}});
        const spy = jest.spyOn(wrapper.vm, 'applyChanges');
        await wrapper.vm.applyChanges();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('mounted component', () => {
        ParentScenario.mounted.call(wrapper.vm);
    });
});

describe('Item Scenario', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };
    const store = new Vuex.Store({
        actions: {
            setListScenarios({commit}, value) {},
            setScenario({commit}, value) {},
            setShowConfig({commit}, value) {},
            setItemScenario({commit}, value) {},
            setChangeNameScenario({commit}, value) {},
            changeItemScenarios({commit}, value) {}
        },
        getters: {
            getListScenarios: () => mock.getListScenarios,
            getScenario: () => mock.getScenario,
            getShowConfig: () => mock.getShowConfig,
            getInitialList: () => mock.getInitialList,
            theme: () => 'dark',
            getItemScenario: () => mock.getItemScenario,
            getUserScenario: () => mock.getUserScenario
        }
    });

    const wrapper = shallowMount(Item, { localVue, router, store, stubs: {Item: true} });
    it('it should  Vue instance', () => {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('getTitleItem toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'getTitleItem');
        await wrapper.vm.getTitleItem({schema: 'vnumber-in'});
        expect(spy).toHaveBeenCalled();
        expect(spy).toHaveReturnedWith({ text: 'V-Numbers', icon: wrapper.vm.icon.vnumberInBlack });
        spy.mockRestore();
    });
    it('getTitleItem toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'getTitleItem');
        await wrapper.vm.getTitleItem({schema: 'queue', option: {name: 'q', number: '1111'}});
        expect(spy).toHaveBeenCalled();
        expect(spy).toHaveReturnedWith({ text: 'q (1111)', icon: wrapper.vm.icon.queueIconBlack });
        spy.mockRestore();
    });
    it('getTitleItem toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'getTitleItem');
        await wrapper.vm.getTitleItem({schema: 'dial', option: {number: '123'}});
        expect(spy).toHaveBeenCalled();
        expect(spy).toHaveReturnedWith({ text: 'AAA (123)', icon: wrapper.vm.icon.dialBlack });
        spy.mockRestore();
    });
    it('getTitleItem toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'getTitleItem');
        await wrapper.vm.getTitleItem({});
        expect(spy).toHaveBeenCalled();
        expect(spy).toHaveReturnedWith({ text: '', icon: '' });
        spy.mockRestore();
    });
    it('chooseItem toHaveBeenCalled', async () => {
        await wrapper.setData({counter: 0});
        const spy = jest.spyOn(wrapper.vm, 'chooseItem');
        await wrapper.vm.chooseItem('change', mock.getItemScenario);
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('chooseItem toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'chooseItem');
        await wrapper.vm.chooseItem('change', {schema: 'dial', option: {number: '123'}});
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('chooseItem toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'chooseItem');
        await wrapper.vm.chooseItem('add', {schema: 'dial', option: {number: '123'}});
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
});

describe('dialConfiguration Scenario', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };

    const store = new Vuex.Store({
        actions: {
            setListScenarios({commit}, value) {},
            setScenario({commit}, value) {},
            setShowConfig({commit}, value) {},
            setItemScenario({commit}, value) {},
            setChangeNameScenario({commit}, value) {},
            changeItemScenarios({commit}, value) {}
        },
        getters: {
            getListScenarios: () => mock.getListScenarios,
            getScenario: () => mock.getScenario,
            getShowConfig: () => mock.getShowConfig,
            getInitialList: () => mock.getInitialList,
            theme: () => 'dark',
            getItemScenario: () => mock.getItemScenarioDial,
            getUserScenario: () => mock.getUserScenario,
        }
    });

    const wrapper = shallowMount(dialConfiguration, { localVue, router, store, dialConfiguration: {Item: true} });
    it('it should  Vue instance', () => {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('viewType toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'viewType');
        await wrapper.vm.viewType('queue');
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.edit).toStrictEqual(false);
        spy.mockRestore();
    });
    it('hide toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'hide');
        await wrapper.vm.hide();
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.showOption).toStrictEqual('');
        spy.mockRestore();
    });
    it('setOptions toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setOptions');
        await wrapper.vm.setOptions('user', mock.getUserScenario[0]);
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('deleteItem toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'deleteItem');
        await wrapper.vm.deleteItem();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setUser toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setUser');
        await wrapper.vm.setUser();
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.user).toStrictEqual({internalNumber: '123', name: 'AAA'});
        spy.mockRestore();
    });
    it('mounted component', () => {
        dialConfiguration.mounted.call(wrapper.vm);
    });
});

describe('queueConfiguration Scenario', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };
    const store = new Vuex.Store({
        actions: {
            setListScenarios({commit}, value) {},
            setScenario({commit}, value) {},
            setShowConfig({commit}, value) {},
            setItemScenario({commit}, value) {},
            setChangeNameScenario({commit}, value) {},
            changeItemScenarios({commit}, value) {}
        },
        getters: {
            getListScenarios: () => mock.getListScenarios,
            getScenario: () => mock.getScenario,
            getShowConfig: () => mock.getShowConfig,
            getInitialList: () => mock.getInitialList,
            theme: () => 'dark',
            getItemScenario: () => mock.getItemScenarioQueue,
            getUserScenario: () => mock.getUserScenario,
            getQueuesScenarios: () => mock.getQueuesScenarios
        }
    });

    const wrapper = shallowMount(queueConfiguration, { localVue, router, store, queueConfiguration: {Item: true} });
    it('it should  Vue instance', () => {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('setOptions toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setOptions');
        await wrapper.vm.setOptions('time', null);
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.minSec).toStrictEqual('sec');
        spy.mockRestore();
    });
    it('setOptions toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setOptions');
        await wrapper.vm.setOptions('queue', mock.getQueuesScenarios);
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.edit).toStrictEqual(true);
        expect(wrapper.vm.queue).toStrictEqual(mock.getQueuesScenarios);
        spy.mockRestore();
    });
    it('hide toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'hide');
        await wrapper.vm.hide();
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.showOption).toStrictEqual('');
        spy.mockRestore();
    });
    it('viewType toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'viewType');
        await wrapper.vm.viewType('dial');
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.edit).toStrictEqual(false);
        spy.mockRestore();
    });
    it('deleteItem toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'deleteItem');
        await wrapper.vm.deleteItem();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setChanges toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setChanges');
        await wrapper.vm.setChanges();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setQueue toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setQueue');
        await wrapper.vm.setQueue();
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.queue).toStrictEqual(mock.getQueuesScenarios);
        expect(wrapper.vm.maxWait).toStrictEqual(3600);
        spy.mockRestore();
    });
    it('mounted component', () => {
        queueConfiguration.mounted.call(wrapper.vm);
    });
});

describe('vNumberConfiguration Scenario', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };
    const store = new Vuex.Store({
        actions: {
            setListScenarios({commit}, value) {},
            setScenario({commit}, value) {},
            setShowConfig({commit}, value) {},
            setItemScenario({commit}, value) {},
            setChangeNameScenario({commit}, value) {},
            changeItemScenarios({commit}, value) {}
        },
        getters: {
            getListScenarios: () => mock.getListScenarios,
            getScenario: () => mock.getScenario,
            getShowConfig: () => mock.getShowConfig,
            getInitialList: () => mock.getInitialList,
            theme: () => 'dark',
            getItemScenario: () => mock.getItemScenario,
            getUserScenario: () => mock.getUserScenario,
            getQueuesScenarios: () => mock.getQueuesScenarios,
            getExternalNumbers: () => mock.getExternalNumbers,
            getCountriesScenario: () => mock.getCountriesScenario
        }
    });

    const wrapper = shallowMount(vNumberConfiguration, { localVue, router, store, vNumberConfiguration: {Item: true} });
    it('it should  Vue instance', () => {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('delVNumber toHaveBeenCalled', async () => {
        await wrapper.setData({selectedNumbers: ["380443640000"]});
        const spy = jest.spyOn(wrapper.vm, 'delVNumber');
        await wrapper.vm.delVNumber("380443640000");
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.selectedNumbers).toStrictEqual([]);
        spy.mockRestore();
    });
    it('toggleVisible toHaveBeenCalled', async () => {
        await wrapper.setData({showOption: 'left'});
        const spy = jest.spyOn(wrapper.vm, 'toggleVisible');
        await wrapper.vm.toggleVisible('left');
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.showOption).toStrictEqual('');
        spy.mockRestore();
    });
    it('toggleVisible toHaveBeenCalled', async () => {
        await wrapper.setData({showOption: 'left'});
        const spy = jest.spyOn(wrapper.vm, 'toggleVisible');
        await wrapper.vm.toggleVisible('right');
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.showOption).toStrictEqual('right');
        spy.mockRestore();
    });
    it('addVNumber toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'addVNumber');
        await wrapper.vm.addVNumber("380443640000");
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.selectedNumbers).toStrictEqual(["380443640000"]);
        spy.mockRestore();
    });
    it('setFiltersParam toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setFiltersParam');
        await wrapper.vm.setFiltersParam('allNumbers', false, 'array', mock.getCountriesScenario);
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setFiltersParam toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setFiltersParam');
        await wrapper.vm.setFiltersParam('allNumbers', "Afghanistan", 'array', mock.getCountriesScenario);
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setFiltersParam toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setFiltersParam');
        await wrapper.vm.setFiltersParam('selectedNumbers', 'begin', 'object');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('clearFilters toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'clearFilters');
        await wrapper.vm.clearFilters('allNumbers');
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.filterParams.allNumbers.vNumber).toStrictEqual({key: 'begin', value: ''});
        expect(wrapper.vm.filterParams.allNumbers.countries).toStrictEqual([]);
        expect(wrapper.vm.filterParams.allNumbers.description).toStrictEqual('');
        expect(wrapper.vm.filterParams.allNumbers.status).toStrictEqual([]);
        spy.mockRestore();
    });
    it('filterNumber toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'filterNumber');
        await wrapper.vm.filterNumber({key: 'begin', value: '38'}, "380443640000");
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('filterNumber toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'filterNumber');
        await wrapper.vm.filterNumber({key: 'contain', value: '38'}, "380443640000");
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('filterNumber toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'filterNumber');
        await wrapper.vm.filterNumber({key: 'end', value: '38'}, "380443640000");
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('filterNumber toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'filterNumber');
        await wrapper.vm.filterNumber({key: 'equal', value: '38'}, "380443640000");
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('hide toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'hide');
        await wrapper.vm.hide();
        expect(spy).toHaveBeenCalled();
        expect(wrapper.vm.showOption).toStrictEqual('');
        spy.mockRestore();
    });
    it('hideListOptions toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'hideListOptions');
        await wrapper.vm.hideListOptions();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    // it('addFlagToNumber toHaveBeenCalled', async () => {
    //     const spy = jest.spyOn(wrapper.vm, 'addFlagToNumber');
    //     await wrapper.vm.addFlagToNumber();
    //     expect(spy).toHaveBeenCalled();
    //     expect(wrapper.vm.edit).toStrictEqual(true);
    //     expect(wrapper.vm.queue).toStrictEqual(mock.getQueuesScenarios);
    //     spy.mockRestore();
    // })
    it('computed agentsFiltered', () => {
        vNumberConfiguration.computed.getCountries.get.call(wrapper.vm);
    });
    // it('computed agentsFiltered', async () => {
    //     await wrapper.setData({
    //         selectedNumbers: [{description: '', country: 'Ukraine'}], 
    //         getExternalNumbers: [{description: '', country: 'Ukraine'}]
    //     })
    //     vNumberConfiguration.computed.selectedNumbersFilter.call(wrapper.vm)
    // })
    it('computed agentsFiltered', async () => {
        await wrapper.setData({
            selectedNumbers: [{description: '', country: 'Ukraine', number: '0880640822'}], 
            getExternalNumbers: [{description: '', country: 'Ukraine', number: '0880640822'}]
        });
        vNumberConfiguration.computed.relatedNumbersFilter.call(wrapper.vm);
    });
    it('mounted component', () => {
        vNumberConfiguration.mounted.call(wrapper.vm);
    });
});