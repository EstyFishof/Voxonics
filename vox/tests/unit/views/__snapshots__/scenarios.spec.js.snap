// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Scenario Check snapshot 1`] = `<!---->`;

exports[`Scenario Check snapshot 1`] = `
<div
  class="wrapper"
>
  <div
    class="header"
  >
    <ul
      class="top-tabs"
    >
      <li
        class="new"
      >
        <span
          class="new-scenario"
          content="Add new scenario"
          tabindex="0"
        >
          <img
            alt="new"
            src="@/assets/img/icons/new-scenario.svg"
          />
        </span>
      </li>
       
      <li
        class="parent-tab active"
      >
        <input
          class="item-input"
          readonly="readonly"
          type="text"
        />
         
        <!---->
         
        <div
          class="bottom-tabs-block"
        >
          <ul
            class="bottom-tabs"
          >
             
            <!---->
          </ul>
        </div>
      </li>
       
      <!---->
    </ul>
     
    <!---->
     
    <div
      class="secondary-header"
    />
  </div>
   
  <div
    class="body"
  >
    <div
      class="list-items"
    >
      <item-stub
        counter="0"
        link="[object Object]"
      />
    </div>
     
    <queueconfiguration-stub
      class="configuration"
    />
  </div>
</div>
`;

exports[`dialConfiguration Scenario Check snapshot 1`] = `
<div
  class="configuration-block"
>
  <div
    class="header"
  >
    <div
      class="left-sector"
    >
      <img
        class="arrow"
        src="@/assets/img/icons/configuration-arrow.svg"
      />
       
      <span
        class="header-title"
      >
        Dial Configuration
      </span>
       
      <img
        class="delete-item"
        content="Delete"
        src="@/assets/img/icons/detele-item-scenario.svg"
        tabindex="0"
      />
    </div>
  </div>
   
  <div
    class="body"
  >
    <span
      style="display: flex;"
    >
      <div
        class="options-block"
      >
        <label
          class="options-label"
        >
          <span
            class="options-title"
          >
            Select dial type
          </span>
           
          <input
            class="input-field num"
            readonly="readonly"
            type="text"
          />
           
          <span
            class="triangle"
          >
            <img
              alt="select"
              src="../../../../assets/img/icons/queues-select-arrow.png"
            />
          </span>
           
          <ul
            class="list-related-numbers"
            style="display: none;"
          >
            <li>
              Queue
            </li>
             
            <li
              class="active-item"
            >
              Agent
            </li>
          </ul>
        </label>
      </div>
       
      <div
        class="options-block"
      >
        <label
          class="options-label"
        >
          <span
            class="options-title"
          >
            Select Agent/Queue
          </span>
           
          <input
            class="input-field num"
            readonly="readonly"
            type="text"
          />
           
          <span
            class="triangle"
          >
            <img
              alt="select"
              src="../../../../assets/img/icons/queues-select-arrow.png"
            />
          </span>
           
          <ul
            class="list-related-numbers"
            style="display: none;"
          >
            <li
              class="active-item"
            >
              AAA 123
            </li>
          </ul>
        </label>
      </div>
    </span>
  </div>
</div>
`;

exports[`queueConfiguration Scenario Check snapshot 1`] = `
<div
  class="configuration-block"
>
  <div
    class="header"
  >
    <div
      class="left-sector"
    >
      <img
        class="arrow"
        src="@/assets/img/icons/configuration-arrow.svg"
      />
       
      <span
        class="header-title"
      >
        Queue Configuration
      </span>
       
      <img
        class="delete-item"
        content="Delete"
        src="@/assets/img/icons/detele-item-scenario.svg"
        tabindex="0"
      />
    </div>
  </div>
   
  <div
    class="body"
  >
    <div
      class="options-block"
    >
      <label
        class="options-label"
      >
        <span
          class="options-title"
        >
          Select dial type
        </span>
         
        <input
          class="input-field num"
          readonly="readonly"
          type="text"
        />
         
        <span
          class="triangle"
        >
          <img
            alt="select"
            src="../../../../assets/img/icons/queues-select-arrow.png"
          />
        </span>
         
        <ul
          class="list-related-numbers"
          style="display: none;"
        >
          <li
            class="active-item"
          >
            Queue
          </li>
           
          <li>
            Agent
          </li>
        </ul>
      </label>
    </div>
     
    <div
      class="options-block"
    >
      <label
        class="options-label"
      >
        <span
          class="options-title"
        >
          Please select a queue
        </span>
         
        <input
          class="input-field num"
          readonly="readonly"
          type="text"
        />
         
        <span
          class="triangle"
        >
          <img
            alt="select"
            src="../../../../assets/img/icons/queues-select-arrow.png"
          />
        </span>
         
        <ul
          class="list-related-numbers"
          style="display: none;"
        >
          <li
            class="active-item"
          >
            AAA ( 1569 )
          </li>
        </ul>
      </label>
    </div>
     
    <div
      class="options-block"
    >
      <label
        class="options-label"
      >
        <span
          class="options-title"
        >
          Please select a dial time
        </span>
         
        <span
          class="time-block"
        >
          <input
            class="input-field time"
            maxlength="3"
            type="text"
          />
           
          <span
            class="minSec"
          >
            min
          </span>
        </span>
      </label>
    </div>
  </div>
</div>
`;

exports[`vNumberConfiguration Scenario Check snapshot 1`] = `
<div
  class="configuration-block"
>
  <div
    class="header"
  >
    <div
      class="left-sector"
    >
      <img
        class="arrow"
        src="@/assets/img/icons/configuration-arrow.svg"
      />
       
      <span
        class="header-title"
      >
        V-Number Configuration
      </span>
    </div>
     
    <!---->
  </div>
   
  <div
    class="body"
  >
    <div
      class="left-sector sector"
    >
      <div
        class="filters"
      >
        <div
          class="options-block"
        >
          <label
            class="options-label"
          >
            <span
              class="options-title"
            >
              V-Number
            </span>
             
            <div
              class="vnumber-block"
            >
              <input
                class="input-field type-vnumber"
                readonly="readonly"
                type="text"
              />
               
              <span
                class="triangle"
              >
                <img
                  alt="select"
                  src="../../../../assets/img/icons/queues-select-arrow.png"
                />
              </span>
               
              <ul
                class="list-related-numbers vnumber-list"
                style="display: none;"
              >
                <li>
                  
                    begin
                  
                </li>
                <li>
                  
                    contain
                  
                </li>
                <li>
                  
                    end
                  
                </li>
                <li>
                  
                    equal
                  
                </li>
              </ul>
               
              <input
                class="input-field enter-vnumber"
                placeholder="Enter numbers"
                type="text"
              />
            </div>
          </label>
        </div>
         
        <div
          class="options-block country-block"
        >
          <label
            class="options-label"
          >
            Countries:
              
            <input
              class="input-field"
              placeholder="Enter Country"
              type="text"
            />
             
            <span
              class="triangle"
            >
              <img
                alt="select"
                src="@/assets/img/icons/queues-select-arrow.png"
              />
            </span>
             
            <ul
              class="list-related-numbers"
              style="display: none;"
            >
              <li>
                All
              </li>
               
              <li
                class=""
              >
                Afghanistan
                
              </li>
            </ul>
          </label>
        </div>
         
        <div
          class="options-block description-block"
        >
          <label
            class="options-label"
          >
            <span
              class="options-title"
            >
              Description
            </span>
             
            <input
              class="input-field"
              placeholder="Enter Description"
              type="text"
            />
          </label>
        </div>
         
        <div
          class="options-block status-block"
        >
          <label
            class="options-label"
          >
            <span
              class="options-title"
            >
              Status
            </span>
             
            <input
              class="input-field"
              placeholder="Used"
              readonly="readonly"
              type="text"
            />
             
            <span
              class="triangle"
            >
              <img
                alt="select"
                src="../../../../assets/img/icons/queues-select-arrow.png"
              />
            </span>
             
            <ul
              class="list-related-numbers status-list"
              style="display: none;"
            >
              <li
                class=""
              >
                Used
              </li>
               
              <li
                class=""
              >
                Not user
              </li>
            </ul>
          </label>
        </div>
         
        <span
          class="clear-btn"
        >
          <img
            src="@/assets/img/icons/blue-broom.svg"
          />
        </span>
      </div>
       
      <div
        class="vnumbers"
      >
        <ul
          class="vnumbers-list"
        >
          <li
            class="vnumber-item"
          >
            <img
              class="flag"
              src="http://23.88.11.75/country/iso2/AF.svg"
            />
             
            <span
              class="vnumber-number"
            >
              
                380443640000
                
              <span
                class="vnumber-description"
              >
                test
              </span>
            </span>
             
            <img
              class="vnumber-btn"
              src="@/assets/img/icons/plus-numbers.svg"
            />
          </li>
        </ul>
      </div>
    </div>
     
    <span
      class="arrows"
    >
      <img
        alt="arrows"
        src="@/assets/img/icons/arrows-scenario.svg"
      />
    </span>
     
    <div
      class="right-sector sector"
    >
      <div
        class="vnumbers"
      >
        <ul
          class="vnumbers-list"
        />
      </div>
       
      <div
        class="filters right-filters"
      >
        <div
          class="options-block"
        >
          <label
            class="options-label"
          >
            <span
              class="options-title"
            >
              V-Number
            </span>
             
            <div
              class="vnumber-block"
            >
              <input
                class="input-field type-vnumber"
                readonly="readonly"
                type="text"
              />
               
              <span
                class="triangle"
              >
                <img
                  alt="select"
                  src="../../../../assets/img/icons/queues-select-arrow.png"
                />
              </span>
               
              <ul
                class="list-related-numbers vnumber-list"
                style="display: none;"
              >
                <li>
                  begin
                  
                </li>
                <li>
                  contain
                  
                </li>
                <li>
                  end
                  
                </li>
                <li>
                  equal
                  
                </li>
              </ul>
               
              <input
                class="input-field enter-vnumber"
                placeholder="Enter numbers"
                type="text"
              />
            </div>
          </label>
        </div>
         
        <div
          class="options-block country-block"
        >
          <label
            class="options-label"
          >
            Countries:
              
            <input
              class="input-field"
              placeholder="Enter Country"
              type="text"
            />
             
            <span
              class="triangle"
            >
              <img
                alt="select"
                src="@/assets/img/icons/queues-select-arrow.png"
              />
            </span>
             
            <ul
              class="list-related-numbers"
              style="display: none;"
            >
              <li>
                All
              </li>
               
              <li
                class=""
              >
                Afghanistan
                
              </li>
            </ul>
          </label>
        </div>
         
        <div
          class="options-block description-block"
        >
          <label
            class="options-label"
          >
            <span
              class="options-title"
            >
              Description
            </span>
             
            <input
              class="input-field"
              placeholder="Enter Description"
              type="text"
            />
          </label>
        </div>
         
        <span
          class="clear-btn"
        >
          <img
            src="@/assets/img/icons/blue-broom.svg"
          />
        </span>
      </div>
    </div>
  </div>
   
  <confirmation-stub
    confirmbtntext="Confirm"
    textconfirmation="This number is used in the \\"<b>undefined</b>\\" route. V-number can be used only in one incoming routing. Are you sure you want to use this V-number here?"
  />
</div>
`;
